### PR TITLE
Delete default vhost as well as www.gov.uk

### PR DESCRIPTION
**Pull request by mattbostock**
For discussion:

This PR will fix the `inicident.fail_to_mirror` task, which currently fails using the default Fabric configuration to fail early on errors. I've tested this on Staging.

I think this failure highlights that deleting individual vhosts may be too brittle, so it might be preferable just to stop the nginx service. Suggestions welcome.

* * *

Delete the default Nginx virtual host as well as the `www.gov.uk`
virtual host.

The default virtual host references the `varnish` upstream which is
defined in the `www.gov.uk` virtual host, meaning that nginx fails to
start if the `www.gov.uk` is not present because it can't find the
`varnish` backend.

This means that the Fabric task `incident.fail_to_mirror` only stops
Nginx on the first cache node; Fabric will not continue to the other
cache nodes because `sudo service nginx start` returns a non-zero exit
code.

It looks like this has been broken since
https://github.gds/gds/puppet/commit/baef41223501337005f1efa4342e01fffaa0139c.

### CODE REVIEW

**Comment by ajlanghorn** [comment]
If we stop Nginx, we're no longer going to be responding on 80 or 443, so the Fastly VCL would force requests through to the mirrors. I think I have that right, but I might not do. So, assuming that is correct, then doing anything but stopping the Nginx service and disabling Puppet seems superfluous?

TL;DR: I agree with Matt.

**Comment by mattbostock** [comment]
If the virtual host doesn't exist, Nginx will return a HTTP 444 (No content) and immediately drop the connection. So that'd force the backend probe to detect that the origin is sick, and trigger the failover.

Stopping nginx would have the same effect.

**Comment by ajlanghorn** [comment]
Sure, so both have the same effect, right? Doing both seems overkill.

**Comment by dcarley** [comment]
Don't have an opinion, but worth noting that we did previously stop Nginx and then later changed the behaviour to be selective:

- e83c0c75184ea0949aedeeef18dfcd9176b9c8de
- 338b44913bed4c223603dbc3b27ef0d76e2ef6d8

**Comment by jennyd** [comment]
Does the [earlier reason for deleting the virtual host](https://github.com/alphagov/fabric-scripts/pull/21#issuecomment-31009889) still apply?

**Comment by mattbostock** [comment]
Top sleuthing @jennyd, I found 338b449 but couldn't fathom the reason for the change.

We now have failover for assets:

https://github.gds/gds/cdn-configs/blob/fb70e873a65b2728c31050b02c341c647d7c95d6/fastly/vcl_templates/assets.vcl.erb#L150-L158

So I think reverting to e83c0c7 would make sense.

**Comment by mattbostock** [comment]
I'll update my PR accordingly.

**Comment by mattbostock** [comment]
Superseded by https://github.com/alphagov/fabric-scripts/pull/133
